x-pack/plugins/uptime/public/components/monitor/synthetics/waterfall/components/waterfall_chart.tsx

```diff
@@ -174,44 +162,47 @@ export const WaterfallChart = ({
           style={{ paddingTop: '10px' }}
           ref={chartWrapperDivRef}
         >
-          {shouldRenderSidebar && (
-            <Sidebar items={sidebarItems!} height={generatedHeight} render={renderSidebarItem!} />
-          )}
+          {shouldRenderSidebar && <Sidebar items={sidebarItems!} render={renderSidebarItem!} />}
           <EuiFlexItem grow={shouldRenderSidebar ? MAIN_GROW_SIZE : true}>
-            <WaterfallChartChartContainer height={generatedHeight}>
-              <Chart className="data-chart">
-                <Settings
-                  showLegend={false}
-                  rotation={90}
-                  tooltip={{ customTooltip: Tooltip }}
-                  theme={theme}
-                />
-
-                <Axis
-                  id="time"
-                  position={Position.Top}
-                  tickFormat={tickFormat}
-                  domain={domain}
-                  showGridLines={true}
-                  style={{
-                    axisLine: {
-                      visible: false,
-                    },
-                  }}
-                />
-
-                <BarSeries
-                  id="waterfallItems"
-                  xScaleType={ScaleType.Linear}
-                  yScaleType={ScaleType.Linear}
-                  xAccessor="x"
-                  yAccessors={['y']}
-                  y0Accessors={['y0']}
-                  styleAccessor={barStyleAccessor}
-                  data={data}
-                />
-              </Chart>
-            </WaterfallChartChartContainer>
+            {chartsToDisplay.map((chartData, ind) => (
+              <WaterfallChartChartContainer
+                height={getChartHeight(chartData, ind)}
```

Comment:
Does this need a key for rendering the list of WaterfallChartChartContainers?

Reply:
yes indeed, i have added that.